huaweicloud/services/nat/resource_huaweicloud_nat_private_snat_rule.go

```diff
@@ -132,6 +142,8 @@ func resourcePrivateSnatRuleRead(_ context.Context, d *schema.ResourceData, meta
 		d.Set("cidr", resp.Cidr),
 		d.Set("created_at", resp.CreatedAt),
 		d.Set("updated_at", resp.UpdatedAt),
+		d.Set("enterprise_project_id", resp.EnterpriseProjectId),
+		d.Set("transit_ip_address", resp.TransitIpAssociations[0].Address),
```

Comment:
```suggestion
		d.Set("transit_ip_address", utils.PathSearch("[0].Address", resp.TransitIpAssociations, nil)),
```
If no transit IP address is associated, this code will cause array subscripts to out-of-bounds.

Reply:
done